lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4671,7 +4685,12 @@ ParseResult FIRCircuitParser::parseToplevelDefinition(CircuitOp circuit,
       return failure();
     return parseLayer(circuit);
   case FIRToken::kw_module:
-    return parseModule(circuit, indent);
+    return parseModule(circuit, /*isPublic=*/false, indent);
+  case FIRToken::kw_public:
+    if (requireFeature({4, 0, 0}, "public modules"))
+      return failure();
+    consumeToken();
```

Comment:
Please adopt something like the code from #6176 here to provide a diagnostic for use of public and add the test case from that PR that presently crashes here:
```
;// -----
FIRRTL version 4.0.0
circuit PublicNonModule:
  ; expected-error @below {{only modules may be public}}
  public extmodule Foo:
  module PublicNonModule:
```